### ISSUE TITLE
🐛 [?1h= before file name

### ISSUE THREAD

**Issue by mvolkmann** [dandavison/delta#1414]
See the red arrow in this screenshot: ![Screenshot 2023-05-10 at 8 16 30 AM](https://github.com/dandavison/delta/assets/79312/879ba0e8-37ed-4863-a3dd-a3ca7bfa8a1a)

Is there some configuration I need to add to fix that?

**Comment by mvolkmann** [comment]
I should have mentioned that I am using delta as my pager inside lazygit.

**Comment by dandavison** [comment]
Sorry, I didn't mean to close the issue in a jarring way! If it is a delta bug, then it should be reproducible by doing the operation that lazygit was doing, but directly from the command line. It says "staged changes" so perhaps it was doing `git diff --cached`?? If you can reproduce it, then sending the raw diff output that causes the bug would be extremely helpful. But, I could definitely be wrong, but I would say most likely a lazygit bug since I think this would have been reported already in delta's issues.

**Comment by mvolkmann** [comment]
@dandavison Naturally the lazygit team thinks it is a delta issue. ;-)
https://github.com/jesseduffield/lazygit/issues/2621

I'm not familiar enough with the internals of lazygit to know how it calls delta.

I also have delta configured in my `.gitconfig` file and when I run "git diff" I do not see those extra characters.
So I suspect you are correct that it is an issue in lazygit.

**Comment by dandavison** [comment]
> I'm not familiar enough with the internals of lazygit to know how it calls delta.

How's your shell-scripting? How about replacing the real delta with a script that writes its arguments and input to disk? Something like

```bash
echo "delta args: $@" > /tmp/delta-input
echo "delta stdin:" >> /tmp/delta-input
cat >> /tmp/delta-input
```

`chmod +x` that and place it somewhere like `~/bin/delta` or somewhere else that's high in your `$PATH`.

**Comment by mvolkmann** [comment]
Thanks! I followed your instructions. Here is what I see in the `/tmp/delta-input` file. There was one modified file in the repository (README.md) and the only change was to add the line "test" at the end.

```
delta args:
delta stdin:
diff --git a/README.md b/README.md
index ec42f67c..53d07cb9 100644
--- a/README.md
+++ b/README.md
@@ -1,4 +1,4 @@
 # ArchDesign

 Please see the client and server folders for details.
-
+test
```

**Comment by dandavison** [comment]
OK, if you delete the first two lines so that only the diff remains, and then

```
cat /tmp/delta-input | delta
```
Do you see the strange characters? If you do then it's a bug in the version of delta you have installed, or a problem with your local delta config etc. If you don't, then that suggests a bug in lazygit. Delta output looks good to me when I do this.

Another thing to check is whether you see the funny characters in lazygit when delta is simple the above script, or a script containing `echo hello` or something. That would obviously disprove a bug in delta also.

**Comment by mvolkmann** [comment]
I do not see strange characters when I run the `cat` command you suggested above. So I think you are correct that the issue is not caused by delta. I learned that `[?1h` is an ANSI escape sequence for positioning the cursor. I'm pretty sure lazygit is outputting that. Thanks so much for walking me through this!

**Comment by dandavison** [comment]
Yep, agreed, an ANSI sequence like that sounds much more like something a TUI would use; delta isn't interacting with the terminal emulator like that.

To demonstrate to the lazygit team that it's not the pager, perhaps you can reproduce it with the pager set to a dummy shell script.
